src/software/ai/passing/eighteen_zone_pitch_division.cpp

```diff
@@ -0,0 +1,49 @@
+#include "software/ai/passing/eighteen_zone_pitch_division.h"
+
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/point.h"
+#include "software/geom/rectangle.h"
+
+EighteenZonePitchDivision::EighteenZonePitchDivision(const Field& field)
+{
+    const double zone_width  = field.xLength() / 6;
+    const double zone_height = field.yLength() / 3;
+
+    for (double pos_x = -field.xLength() / 2; pos_x < field.xLength() / 2;
+         pos_x += zone_width)
+    {
+        for (double pos_y = field.yLength() / 2; pos_y > -field.yLength() / 2;
+             pos_y -= zone_height)
+        {
+            pitch_division_.emplace_back(Rectangle(
+                Point(pos_x, pos_y), Point(pos_x + zone_width, pos_y - zone_height)));
+        }
+    }
```

Comment:
We could probably use a general function that takes the number of "rows" and "columns" and generates a set of rectangles

Reply:
A pitch division can label zones in any order

if we pulled this into a generic function, the ordering of the rectangles would matter, you would need to sort them after the fact, or have some garuntees on that generic function

 so it seemed like its best to leave this up to the implementation of the pitch division?